Reporting/M4 Document/Summary.md

```diff
@@ -0,0 +1,6 @@
+1. # <a name="_29cp0gkd2ju8"></a>**Summary**
+- For our team Accord, we believe that the project has made significant progress in implementing various functionalities such as registration, login, linking, friend list display, chat history display, server list display, and live messaging. These functionalities have undergone ad-hoc, unit, and black-box testing to ensure their correctness and reliability. However, some features are still under development, notably the state-validation functionality, which impacts the accuracy of user data display.
+- Process:
+  - Our team has adopted a structured approach to testing, utilizing ad-hoc, unit, and black-box testing in the current phase, with plans to incorporate integration testing, acceptance testing, and usability testing in the future. The division of testing approaches aligns well with the development cycle, with ad-hoc testing for quick checks, unit testing for feature validation, and black-box testing for component verification. However, the team may need to ensure better synchronization between testing and development to address ongoing issues with state validation functionality.
```

Comment:
Agreed. A lot of our testing has relied on ad-hoc testing due to the fact that we are using many providers that have their own set of testing tools. For example, a lot of the webhook testing for the database syncing was done by pinging our webhook with sampled data to see if our MongoDB database updates in real-time.